distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Regex/Pattern_2.enso

```diff
@@ -0,0 +1,180 @@
+import project.Any.Any
+#import project.Data.Boolean.Boolean
+from project.Data.Boolean import Boolean, True, False
+import project.Data.Map.Map
+import project.Data.Numbers.Integer
+import project.Data.Range.Extensions
+import project.Data.Range.Range
+import project.Data.Text.Span.Span
+import project.Data.Text.Span.Utf_16_Span
+import project.Data.Text.Regex.Match_2.Match_2
+import project.Data.Text.Regex_2.No_Such_Group
+import project.Data.Text.Text
+import project.Data.Vector.Vector
+import project.Error.Error
+import project.Meta
+import project.Nothing.Nothing
+import project.Panic.Panic
+import project.Polyglot
+
+polyglot java import org.enso.base.Text_Utils
+
+type Pattern_2
+    ## internal_regex_object : RegexObject (Truffle)
+       (See https://github.com/oracle/graal/blob/master/regex/docs/README.md)
+    Value (internal_regex_object : Any)
+
+    matches : Text -> Boolean
+    matches self input =
+        m = self.internal_regex_object.exec input 0
+        m . isMatch && m.getEnd 0 == input.length
+
+    ## PRIVATE
+
+       Tries to match the provided `input` against the pattern `self`.
+
+       Arguments:
+       - input: The text to match the pattern described by `self` against.
+    match : Text -> Match_2 | Nothing
+    match self input =
+        it = Match_Iterator.new self input
+        case it.next of
+            Match_Iterator_Value.Next _ match _ -> match
+            Match_Iterator_Value.Last _ -> Nothing
+
+    ## PRIVATE
+
+       Tries to match the provided `input` against the pattern `self`.
+
+       Returns a Vector of Match_2 objects.
+
+       Arguments:
+       - input: The text to match the pattern described by `self` against.
+    match_all : Text -> Vector Match_2
```

Comment:
We do this all the time around the codebase.

There were at least 3 approaches: `Vector a`, `[a]` and `Vector` (as you suggest, without the type param).

I think it's indeed problematic that we don't have a consensus of if it's right or not.

I appreciate that adding the type to the Vector serves as documentation as we know what kinds of elements are expected there. We rely on this throughout many places, so if anything, I think ideally we'd want to be able to 'officially' have this vector type parameter in (not necessarily now, but I'd prefer to go in the direction where we _can_ have these types rather then removing them and making the type signatures serve less purpose (since they are not checked, they at least serve the documentation purpose)).

Reply:
Yes, I put this in because I saw it used elsewhere.